narwhals/_duckdb/utils.py

```diff
@@ -198,8 +198,11 @@ def narwhals_to_native_dtype(dtype: DType | type[DType], version: Version) -> st
         inner = narwhals_to_native_dtype(dtype.inner, version)  # type: ignore[union-attr]
         return f"{inner}[]"
     if isinstance_or_issubclass(dtype, dtypes.Struct):  # pragma: no cover
-        msg = "todo"
-        raise NotImplementedError(msg)
+        inner = ", ".join(
+            f"{field.name} {narwhals_to_native_dtype(field.dtype, version)}"
+            for field in dtype.fields  # type: ignore[union-attr]
+        )
+        return f"STRUCT({inner})"
```

Comment:
does this work if column names contain spaces? we might need
```
f'"{field.name}" {narwhals_to_native_dtype(field.dtype, version)}'
```
?

Reply:
No, it didn't work, thanks for pointing this out! I added the suggestion and also changed the test to have a column with an empty space